staking/contracts/StakingBase.sol

```diff
@@ -123,14 +123,18 @@ contract StakingBase is OwnableUpgradeable {
      * newAmount=k*amount/ST_FORMULA_DIVIDER
      * newSlope=k*slope/ST_FORMULA_DIVIDER
      **/
-    function getStake(uint amount, uint slope, uint cliff) internal pure returns (uint stakeAmount, uint stakeSlope) {
+    function getStake(uint amount, uint slope, uint cliff) public pure returns (uint stakeAmount, uint stakeSlope) {
         uint cliffSide = cliff.mul(cliff).mul(ST_FORMULA_CLIFF_MULTIPLIER);
 
-        uint slopePeriod = amount.div(slope);
+        uint slopePeriod = divUp(amount, slope);
         uint slopeSide = slopePeriod.mul(slopePeriod).mul(ST_FORMULA_SLOPE_MULTIPLIER);
         uint multiplier = cliffSide.add(slopeSide).add(ST_FORMULA_COMPENSATE).div(ST_FORMULA_MULTIPLIER);
         stakeAmount = amount.mul(multiplier).div(ST_FORMULA_DIVIDER);
-        stakeSlope = slope.mul(multiplier).div(ST_FORMULA_DIVIDER);
+        stakeSlope = divUp(stakeAmount, slopePeriod);
+    }
+
+    function divUp(uint a, uint b) internal pure returns (uint) {
+        return ((a.sub(1)).div(b)).add(1);
```

Comment:
why so? I think that's not correct

Reply:
I check this formula, seems to be correct